src/non_interactive.rs

```diff
@@ -4,21 +4,59 @@ use anyhow::{self, Context};
 use async_std::prelude::StreamExt;
 use async_std::io::{stdin, stdout};
 use async_std::io::prelude::WriteExt;
+use async_std::fs::{read as async_read};
 
 use bytes::BytesMut;
 use edgeql_parser::preparser;
 use edgedb_protocol::value::Value;
 
+use crate::commands::ExitCode;
 use crate::options::Options;
+use crate::options::Query;
+use crate::repl::OutputMode;
 use crate::print::{self, PrintError};
 use edgedb_client::reader::ReadError;
 use crate::statement::{ReadStatement, EndOfFile};
 use edgedb_client::client::Connection;
 use edgedb_client::errors::NoResultExpected;
 use crate::outputs::tab_separated;
 
+use edgeql_parser::preparser::{full_statement};
 
-pub async fn main(options: Options)
+pub async fn main(q: &Query, options: &Options)
+    -> Result<(), anyhow::Error>
+{
+    let mut query_opts = options.clone();
+    query_opts.output_mode = if q.tab_separated {
+        OutputMode::TabSeparated
+    } else if q.json {
+        OutputMode::Json
+    } else {
+        // Let the top-level --json / --tab-seaparated propagate
+        // (they are deprecated, though.)
+        options.output_mode
+    };
+
+    if let Some(file) = &q.file {
+        if file == "-" {
+            interpret_stdin(query_opts).await?;
+        } else {
+            interpret_file(file.to_string(), query_opts).await?;
+        }
+    } else if let Some(queries) = &q.queries {
+        let mut conn = options.create_connector()?.connect().await?;
+        for query in queries {
+            run_query(&mut conn, query, &query_opts).await?;
+        }
+    } else {
+        eprintln!("Either a --file option or \
```

Comment:
Prefix with `error:` for consistency with other such messages?

Reply:
done